src/a2a/server/models.py

```diff
@@ -57,26 +59,30 @@ def __init__(self, pydantic_type: type[T], **kwargs: dict[str, Any]):
     def process_bind_param(
         self, value: T | None, dialect: Dialect
     ) -> dict[str, Any] | None:
-        """Convert Pydantic model to a JSON-serializable dictionary for the database."""
+        """Convert Pydantic model or Protobuf message to a JSON-serializable dictionary for the database."""
         if value is None:
             return None
-        return (
-            value.model_dump(mode='json')
-            if isinstance(value, BaseModel)
-            else value
-        )
+        if isinstance(value, ProtoMessage):
+            return MessageToDict(value, preserving_proto_field_name=False)
+        if isinstance(value, BaseModel):
+            return value.model_dump(mode='json')
+        return value  # type: ignore[return-value]
 
     def process_result_value(
         self, value: dict[str, Any] | None, dialect: Dialect
     ) -> T | None:
-        """Convert a JSON-like dictionary from the database back to a Pydantic model."""
+        """Convert a JSON-like dictionary from the database back to a Pydantic model or Protobuf message."""
         if value is None:
             return None
-        return self.pydantic_type.model_validate(value)
+        # Check if it's a protobuf message class
+        if isinstance(self.pydantic_type, type) and issubclass(self.pydantic_type, ProtoMessage):
+            return ParseDict(value, self.pydantic_type())  # type: ignore[return-value]
+        # Assume it's a Pydantic model
+        return self.pydantic_type.model_validate(value)  # type: ignore[union-attr]
```

Comment:
No need to support Pydantic here IMO, just need to support the protobuf types. 

One thought: We need to understand if we need to support old messages/tasks stored in the DB and translate them to the protobuf types. Otherwise we may require people to drop the DBs for their Agents